pkg/repo/fs.go

```diff
@@ -82,39 +85,43 @@ func (fsr *FsRepo) Version() (int, error) {
 }
 
 // Init initializes a new repo, returning an error if the repo already exists.
-func (fsr *FsRepo) Init(c config.ReadWriter) error {
+func (fsr *FsRepo) Init(cfg types.Bacalhau) error {
 	if exists, err := fsr.Exists(); err != nil {
 		return err
 	} else if exists {
 		return fmt.Errorf("cannot init repo: repo already exists")
 	}
 
-	log.Info().Msgf("Initializing repo at '%s' for environment '%s'", fsr.path, config.GetConfigEnvironment())
+	log.Info().Msgf("Initializing repo at %s", fsr.path)
 
-	// 0755: Owner can read, write, execute. Others can read and execute.
-	if err := os.MkdirAll(fsr.path, repoPermission); err != nil && !os.IsExist(err) {
-		return err
-	}
-
-	cfg, err := c.Current()
+	// if it takes longer than 5 seconds to get the node name from a provider, fail
+	nameCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	nodeName, err := getNodeID(nameCtx, cfg.NameProvider)
 	if err != nil {
 		return err
 	}
```

Comment:
The first time a node is internalized we create its name and write it to the file `system_metadata.yaml` in this method.

Reply:
We still don't persist the installationID in the system_metadata

Reply:
We do in the migration `v3_4.go` if one is present in the config.

In our meeting today it was decided that the installationID will continue to be persisted by the install script, but rather than modify the config file or `system_metadata.yaml` directly it will place it in a file called `~/.config/bacalhau/metadata.yaml` that bacalhau can read from.